hack/test/e2e-aws.sh

```diff
@@ -0,0 +1,185 @@
+
+    exit 1
+  fi
+fi
+
+TAG="${TAG:-$(git describe --tag --always --dirty)}"
+REGION="us-east-1"
+BUCKET="talos-ci-e2e"
+PLATFORM=$(uname -s | tr "[:upper:]" "[:lower:]")
+TALOS_VERSION="${TALOS_DEFAULT:-v0.12.0}"
+K8S_VERSION="${K8S_VERSION:-v1.21.3}"
+KUBECONFIG=
+AMI=
+PROVIDER=aws:v0.6.7
+
+CREATED_CLUSTER=""
+TALOSCTL_PATH="${TMP}/talosctl"
+TALOSCTL="${TALOSCTL_PATH} --talosconfig=${TMP}/talosconfig"
+KUSTOMIZE="${TMP}/kustomize"
+
+cleanup() {
+  if [ "$1" != "0" ]; then
+    # gather container logs
+    if [[ ! -z ${KUBECONFIG} ]]; then
+      curl -Lo kubectl "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/${PLATFORM}/amd64/kubectl"
+      chmod +x kubectl
+
+      ./kubectl logs -n capa-system deployment/capa-controller-manager manager
+      ./kubectl logs -n cacppt-system deployment/cacppt-controller-manager
+    fi
+  fi
+
+  # delete deployed cluster
+  if [[ ! -z ${CREATED_CLUSTER} ]]; then
+    echo "destroying deployed cluster"
+    rm -rf ~/.talos/clusters/${CREATED_CLUSTER}
+    ${TALOSCTL} cluster destroy --name=${CREATED_CLUSTER} || true
+  fi
+
+  rm -rf ${TMP}
+  trap - EXIT
+}
+
+trap 'cleanup $?' INT TERM EXIT
+
+function build_registry_mirrors {
+  if [[ "${CI:-false}" == "true" ]]; then
+    REGISTRY_MIRROR_FLAGS=
+
+    for registry in docker.io k8s.gcr.io quay.io gcr.io ghcr.io registry.dev.talos-systems.io; do
+      local service="registry-${registry//./-}.ci.svc"
+      local addr=`python3 -c "import socket; print(socket.gethostbyname('${service}'))"`
+
+      REGISTRY_MIRROR_FLAGS="${REGISTRY_MIRROR_FLAGS} --registry-mirror ${registry}=http://${addr}:5000"
+    done
+  else
+    # use the value from the environment, if present
+    REGISTRY_MIRROR_FLAGS=${REGISTRY_MIRROR_FLAGS:-}
+  fi
+}
+
+function config {
+  curl -Lo ${TMP}/kustomize.tar.gz https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.1.0/kustomize_v4.1.0_${PLATFORM}_amd64.tar.gz
+  tar -xf ${TMP}/kustomize.tar.gz -C ${TMP} && rm ${TMP}/kustomize.tar.gz
+
+  export INFRASTRUCTURE_COMPONENTS_PATH=${TMP}/control-plane-talos/${TAG}/components.yaml
+  mkdir -p $(dirname ${INFRASTRUCTURE_COMPONENTS_PATH})
+
+  cd config/manager
+  ${KUSTOMIZE} edit set image controller=${REGISTRY_AND_USERNAME}/${NAME}:${TAG}
+  cd -
+  ${KUSTOMIZE} build config >${INFRASTRUCTURE_COMPONENTS_PATH}
+}
+
+function cluster {
+  curl -Lo ${TALOSCTL_PATH} https://github.com/talos-systems/talos/releases/download/${TALOS_VERSION}/talosctl-${PLATFORM}-amd64
+
+  chmod +x ${TALOSCTL_PATH}
+
+  CREATED_CLUSTER="cacppt-test"
+  TAG="${TALOS_VERSION}" ${TALOSCTL} cluster create \
+		--name=${CREATED_CLUSTER} \
+		--kubernetes-version=${K8S_VERSION} \
+    ${REGISTRY_MIRROR_FLAGS} \
+    --crashdump
+  ${TALOSCTL} config nodes 10.5.0.2
+  ${TALOSCTL} kubeconfig -f ${TMP}/kubeconfig
+
+  export KUBECONFIG=${TMP}/kubeconfig
+}
+
+function create_ami {
+  image_name=talos-e2e-cacppt-${TALOS_VERSION}
```

Comment:
we shouldn't need to upload AMIs, as we're running some released version of Talos

https://www.talos.dev/docs/v0.12/cloud-platforms/aws/#official-ami-images